cpp/src/arrow/flight/sql/CMakeLists.txt

```diff
@@ -124,8 +126,15 @@ if(ARROW_BUILD_TESTS OR ARROW_BUILD_EXAMPLES)
                  STATIC_LINK_LIBS
                  ${ARROW_FLIGHT_SQL_TEST_LINK_LIBS}
                  ${ARROW_FLIGHT_SQL_TEST_LIBS}
+                 nanoarrow
+                 AdbcValidation::adbc_validation
+                 # Needs to come twice since the validation library
+                 # also uses ADBC symbols, which get provided by
+                 # libarrow_flight_sql here.
+                 ${ARROW_FLIGHT_SQL_TEST_LINK_LIBS}
```

Comment:
Wouldn't it be enough to instead move `nanoarrow` and `AdbcValidation::adbc_validation` before the Flight SQL link libs? Or am I misunderstanding something?

Reply:
Hmm, indeed. Fixed.